### ISSUE TITLE
Panics from dissimilar::cleanup_merge

### ISSUE THREAD

**Issue by upsuper** [dtolnay/dissimilar#6]
This is a test case: [dissimilar-test.zip](https://github.com/dtolnay/dissimilar/files/5835490/dissimilar-test.zip)

To reproduce, extract the testcase, and run `cargo test --release` (`--debug` is too slow to run).

There are two testcases:

`test_50_000` tests the first 50k bytes, reporting:
```
thread 'tests::test_50_000' panicked at 'slice index starts at 33329 but ends at 33326', /home/upsuper/.cargo/registry/src/github.com-1ecc6299db9ec823/dissimilar-1.0.2/src/range.rs:94:6
```

`test_70_000` tests the first 70k bytes, reporting:
```
thread 'tests::test_70_000' panicked at 'byte index 21992 is not a char boundary; it is inside '䀀' (bytes 21991..21994) of ...
```

**Comment by upsuper** [comment]
I was hoping that #10 can fix this as well, but unfortunately not. There are probably other places deep in the code which doesn't handle character boundary correctly.

**Comment by dtolnay** [comment]
Fixed in 1.0.6.
